lib/rules/block-indentation.js

```diff
@@ -83,16 +85,46 @@ module.exports = class BlockIndentation extends Rule {
 
     switch (configType) {
       case 'number':
-        return config;
+        return {
+          indentation: config,
+        };
       case 'boolean':
-        return config ? 2 : false;
+        if (config) {
+          return {
+            indentation: 2,
+          };
+        } else {
+          return {};
+        }
       case 'string':
         if (config === 'tab') {
-          return 1;
+          return {
+            indentation: 1,
+          };
         }
         break;
+      case 'object': {
+        let result = {
+          indentation: 2,
+        };
+        if ('skip-comments' in config) {
+          let skipComments = config['skip-comments'];
+          if (typeof skipComments === 'boolean') {
+            result.skipComments = skipComments;
+          }
+        }
+        if ('indentation' in config) {
+          let indentation = config.indentation;
+          if (typeof indentation === 'number') {
+            result.indentation = indentation;
+          } else {
+            break;
```

Comment:
Can you add asserts for unexpected config input?
```
const assert = require('assert');

assert(false, 'unexpected config indentation');
```

Reply:
Done